contracts/scripts/ci/generate_uml.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+function check_chainlink_dir() {
+  local param_dir="chainlink"
+  current_dir=$(pwd)
+
+  current_base=$(basename "$current_dir")
+
+  if [ "$current_base" != "$param_dir" ]; then
```

Comment:
Nit but since you're using bash, I'd always use double brackets `[[ ... ]]` for tests. See this [comparison](https://www.baeldung.com/linux/bash-single-vs-double-brackets#:~:text=While%20comparing%20variables%20in%20Bash,successful%20in%20performing%20the%20comparison.).

Reply:
done